File: Data Structures & Algorithms/merge-intervals/submission-1.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        intervals.sort()
        res = []
        prev_end = -1

        for start, end in intervals:
            if prev_end < start:
                prev_end = end
                res.append([start, end])
            else:
                res[-1][1] = max(res[-1][1], end)
                prev_end = res[-1][1]
        
        return res

File: Data Structures & Algorithms/sum-of-two-integers/submission-1.java
class Solution {
    public int getSum(int a, int b) {
        long i = 1;
        int carry = 0;
        int res = 0;
        while(i < 0x100000000L) {
            if((i & a) != 0 && (i & b) != 0) {
                if(carry == 1) {
                    res |= (int)i;
                }
                carry = 1;
            }
            else if((i & a) != 0 || (i & b) != 0) {
                if(carry != 1) {
                    res |= (int)i;
                }
            }
            else {
                if(carry == 1) {
                    res |= (int)i;
                    carry = 0;
                }
            }
            i *= 2;
        }
        return res;
    }
}
